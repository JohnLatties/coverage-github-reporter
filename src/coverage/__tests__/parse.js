/* eslint-env jest */

const { resolve } = require('path')

const { parseFile } = require('../parse')

describe('parseFile', () => {
  it('parses', () => {
    expect(parseFile('/root/app', resolve(__dirname, 'coverage-final.json'))).toMatchSnapshot()
  })
  it('parses 2', () => {
    expect(parseFile('/root/app', resolve(__dirname, 'coverage-final2.json'))).toMatchSnapshot()
  })
  it('parses 3', () => {
    expect(parseFile('/root/app', resolve(__dirname, 'coverage-final3.json'))).toMatchSnapshot()
  })
})
