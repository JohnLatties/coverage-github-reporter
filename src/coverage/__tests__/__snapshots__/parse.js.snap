// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseFile parses 1`] = `
Object {
  "*": Object {
    "htmlRoot": "src/",
    "percent": 18.57142857142857,
  },
  "src/": Object {
    "files": Object {
      "bot.js": Object {
        "percent": 0,
      },
      "cli.js": Object {
        "percent": 0,
      },
      "format-coverage.js": Object {
        "percent": 36.13458528951487,
      },
      "github-comment.js": Object {
        "percent": 0,
      },
      "parse-coverage.js": Object {
        "percent": 13.636363636363637,
      },
    },
    "htmlPath": "",
    "percent": 18.57142857142857,
  },
}
`;

exports[`parseFile parses 2 1`] = `
Object {
  "*": Object {
    "htmlRoot": "src/",
    "percent": 58.705357142857146,
  },
  "src/": Object {
    "files": Object {
      "bot.js": Object {
        "percent": 0,
      },
      "cli.js": Object {
        "percent": 0,
      },
      "format-coverage.js": Object {
        "percent": 94.05320813771519,
      },
      "github-comment.js": Object {
        "percent": 0,
      },
      "parse-coverage.js": Object {
        "percent": 90.58441558441558,
      },
    },
    "htmlPath": "",
    "percent": 58.705357142857146,
  },
}
`;

exports[`parseFile parses 3 1`] = `
Object {
  "*": Object {
    "htmlRoot": "subdir/src/",
    "percent": 54.46241911398706,
  },
  "subdir/src/": Object {
    "files": Object {
      "blank.js": Object {
        "percent": 100,
      },
      "bot.js": Object {
        "percent": 0,
      },
      "cli.js": Object {
        "percent": 0,
      },
      "curl.js": Object {
        "percent": 0,
      },
      "format-coverage.js": Object {
        "percent": 89.55566905005108,
      },
      "github-comment.js": Object {
        "percent": 0,
      },
      "parse-coverage.js": Object {
        "percent": 90.58441558441558,
      },
    },
    "htmlPath": "",
    "percent": 54.46241911398706,
  },
}
`;

exports[`parseFile parses 4 1`] = `
Object {
  "*": Object {
    "htmlRoot": "src/",
    "percent": 58.6637239165329,
  },
  "src/": Object {
    "files": Object {
      "bot.js": Object {
        "percent": 0,
      },
      "cli.js": Object {
        "percent": 0,
      },
      "curl.js": Object {
        "percent": 0,
      },
      "github-comment.js": Object {
        "percent": 0,
      },
    },
    "htmlPath": "",
    "percent": 0,
  },
  "src/coverage/": Object {
    "files": Object {
      "format.js": Object {
        "percent": 100,
      },
      "parse.js": Object {
        "percent": 100,
      },
    },
    "htmlPath": "coverage/",
    "percent": 100,
  },
}
`;
